bip39/src/wordlist/mod.rs

```diff
@@ -33,7 +60,8 @@ pub trait Wordlist {
     /// Returns the index of a given word from the word list.
     fn get_index(word: &str) -> Result<usize, WordlistError> {
         Self::get_all()
-            .binary_search(&word)
-            .map_err(|_| crate::WordlistError::InvalidWord(word.to_string()))
+            .iter()
```

Comment:
are some wordlists not in sorted order? Binary search will be significantly quicker than a linear pass over the list

Reply:
answering my own question. Following clearly shows that the chinese simplified wordlist (at least) is not sorted. Oof.

```
    #[test]
    fn sort() {
        let mut words = vec![
            "的", "一", "是", "在", "不", "了", "有", "和", "人", "这", "中", "大", "为", "上",
        ];
        dbg!(&words);
        words.sort();
        dbg!(&words);
    }
```

Reply:
Would you mind going into each sorted language, and adding an override for `fn get_index(word: &str)` that uses the binary-search? that way we can improve performance on any language that supports it.

That would be English, Italian, Korean, Portugese

```
    fn is_sorted<T>(data: &[T]) -> bool
    where
        T: Ord,
    {
        data.windows(2).all(|w| w[0] <= w[1])
    }

    #[test]
    fn langs_sorted() {
        dbg!(is_sorted(&english::PARSED));
        dbg!(is_sorted(&chinese_simplified::PARSED));
        dbg!(is_sorted(&chinese_traditional::PARSED));
        dbg!(is_sorted(&czech::PARSED));
        dbg!(is_sorted(&french::PARSED));
        dbg!(is_sorted(&italian::PARSED));
        dbg!(is_sorted(&japanese::PARSED));
        dbg!(is_sorted(&korean::PARSED));
        dbg!(is_sorted(&portuguese::PARSED));
        dbg!(is_sorted(&spanish::PARSED));
    }
```

<img width="566" alt="Screenshot 2023-04-02 at 10 46 43 AM" src="https://user-images.githubusercontent.com/10149425/229363844-8f6f9319-ab4d-4c88-b2f2-3b143fca2824.png">